### DISCUSSION TITLE
Preserve Multiline Formatting for Complex Query Chains

### DISCUSSION THREAD

**Discussion by zbambusek** [astral-sh/ruff#14277] [Ideas]
I'm finding that ruff currently condenses multiline chained statements, which can negatively impact readability in the following case:
```python
query = select(DocumentsFiles.files_id)
    .where(DocumentsFiles.files_id.isnot(None))
    .order_by(desc(DocumentsFiles.id))
```
formats to:
```python
query = select(DocumentsFiles.files_id).where(DocumentsFiles.files_id.isnot(None)).order_by(desc(DocumentsFiles.id))
```

**Comment by MichaReiser** [comment]
The example that you shared isn't valid but I suspect you meant 

```py
query = (
    select(DocumentsFiles.files_id)
    .where(DocumentsFiles.files_id.isnot(None))
    .order_by(desc(DocumentsFiles.id))
)
```

Ruff does keep the above formatting if putting the entire chain on a single line exceeds the configured line length.

**Reply by zbambusek** [reply]
Thanks, this was about having an option to set an exception for the alchemy specific queries. 
Reducing line length would indeed resolve the issue, but that would affect the rest of the code.

This is just a suggestion.

**Reply by MichaReiser** [reply]
The issue with never collapsing is that it makes formatting non-reversible which is a rather important property. Reversability
ensures that a change made by the formatter is "undone" when the code changes. In this specific context:

* The formatter inserts the parentheses if the line gets too long
* The reversability guarantees that the formatter removes the parentheses again if you shorten the call chain.
